### PR TITLE
[REVIEW] Reset index drop

### PR DESCRIPTION
**Pull request by mrocklin**
This adds a `to_frame` method and the drop= keyword to `reset_index`

### CODE REVIEW

**Comment by mrocklin** [comment]
Seems to be failing due to https://github.com/rapidsai/cudf/issues/986

File: python/cudf/dataframe/dataframe.py [hunk 1]
```diff
@@ -673,7 +673,11 @@ def set_index(self, index):
                 df[k] = self[k].set_index(index)
             return df
 
-    def reset_index(self):
+    def reset_index(self, drop=False):
+        if not drop:
+            name = self.index.name or 'index'
+            self[name] = self.index
+            self = self[[name] + list(self.columns[:-1])]
```
Comment (kkraus14) [inline python/cudf/dataframe/dataframe.py]: This runs inplace instead of returning by shallow copy.

Comment (mrocklin) [reply python/cudf/dataframe/dataframe.py]: So I should call `.copy()`?

**Comment by mrocklin** [comment]
I'm also not sure how best to handle https://github.com/rapidsai/cudf/issues/986 .  Help there would be welcome.

File: python/cudf/dataframe/dataframe.py (hunk 1 above)
Comment (kkraus14) [reply python/cudf/dataframe/dataframe.py]: > So I should call `.copy()`?

I think your best bet would be to instantiate a new DataFrame object and then follow the logic you have here and it should be good to go.

**Comment by kkraus14** [comment]
> I'm also not sure how best to handle #986 . Help there would be welcome.

I'll tackle that shortly, thanks!

**Comment by mrocklin** [comment]
It looks like I committed a git-submodule change.  My apologies about that.  To be honest I'm not yet sure how to undo this change...

**Comment by kkraus14** [comment]
> It looks like I committed a git-submodule change. My apologies about that. To be honest I'm not yet sure how to undo this change...

Looks like you're just missing the latest changes. If you run a `git submodule update --init --recursive --remote` it should resolve it I believe.

**Pull request merged**